lib/functions.php

```diff
@@ -104,13 +114,22 @@ function rethrow(Promise $promise) {
 /**
  * Runs the event loop until the promise is resolved. Should not be called within a running event loop.
  *
- * @param \Amp\Promise $promise
+ * @param \Amp\Promise|\React\Promise\PromiseInterface $promise
  *
  * @return mixed Promise success value.
  *
+ * @throws \TypeError If $promise is not an instance of \Amp\Promise or \React\Promise\PromiseInterface.
  * @throws \Throwable Promise failure reason.
  */
-function wait(Promise $promise) {
+function wait($promise) {
+    if (!$promise instanceof Promise) {
+        if ($promise instanceof ReactPromise) {
+            $promise = adapt($promise);
+        } else {
+            throw new \TypeError("Must provide an instance of %s or %s", Promise::class, ReactPromise::class);
```

Comment:
I prefer the _typeCheckFailed; that way the class name will still be TypeError in the error log and not Amp\TypeError which may be confusing?